### PR TITLE
Update tests to use test/assert module instead the deprecated TestRunner class (#222)

### PR DESCRIPTION
**Pull request by xabolcs**
Fix for #222.

### CODE REVIEW

File: extension/test/test-logger.js [hunk 1]
```diff
@@ -45,55 +45,53 @@ var verifyAsyncOutput = function (test, logger, testFunc) {
...
-    test.assertEqual(message[1].data.x, 60);
-    test.assertEqual(message[2].data.x, 70);
+  logger.stop(verifyAsyncOutput(assert, done, logger, function (message) {
+    assert.equal(message.length, 3, "Log length is 3");
+    assert.equal(message[0].data.x, 50, "The 1st element is 50 in the log");
+    assert.equal(message[1].data.x, 60, "The 2nd element is 60 in the log");
+    assert.equal(message[2].data.x, 70, "The 3rd element is 70 in the log");
   }));
-
-  test.waitUntilDone(2000);
```
Comment (xabolcs) [inline extension/test/test-logger.js]: I didn't transformed this (and it's friends) to `test/utils/waitUntil` because there is a pending PR about it's removal.
See [Bug 1171551][1]!

[1]: https://bugzilla.mozilla.org/show_bug.cgi?id=1171551 "Remove waitUntil from sdk/test/utils"

**Comment by whimboo** [comment]
The latest Travis run was based on cfx but not jpm. I would suggest that we wait for the other PR #203 to be landed first.

File: extension/test/test-logger.js (hunk 1 above)
Comment (whimboo) [inline extension/test/test-logger.js]: So we don't need an extra wait until the verify method is done? Don't we exit too early in such a case?

Comment (xabolcs) [inline extension/test/test-logger.js]: The second (`done`) argument makes the test asynchronous.

**Comment by xabolcs** [comment]
PR updated with `jpm`.

File: extension/test/test-logger.js (hunk 1 above)
Comment (whimboo) [inline extension/test/test-logger.js]: The whole nesting makes it kinda hard to follow the code. I hope that we can soon convert all those async calls into nice to read promise chains.

**Comment by whimboo** [comment]
Looks all fine to me. The remaining failure for esr38 is covered by another PR.  So you can get this landed if you can.

**Comment by xabolcs** [comment]
Landed as 3447f425e0e9d4419bf4cc804c03f9b719fcfb4f.

File: extension/test/test-logger.js (hunk 1 above)
Comment (xabolcs) [inline extension/test/test-logger.js:93]: > I hope that we can soon convert all those async calls into nice to read promise chains.

Does #219 cover that or it needs a separate issue?

Comment (whimboo) [inline extension/test/test-logger.js:93]: That should be fine, yes.
